### PR TITLE
DB-11345 Increase frequency sketch max map size for FK columns

### PR DESCRIPTION
**Pull request by ascend1**

### CODE REVIEW

**Comment by martinrupp** [comment]
please provide some description what this does and how to test is. or is this just a rebase?

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0

File: db-engine/src/main/java/com/splicemachine/db/iapi/stats/ColumnStatisticsImpl.java [hunk 1]
```diff
@@ -111,6 +151,7 @@ public ColumnStatisticsImpl(DataValueDescriptor dvd,
     public void writeExternal(ObjectOutput out) throws IOException {
         out.writeLong(nullCount);
         out.writeObject(dvd);
+        out.writeInt(frequenciesSketchMaxMapSize);
```
Comment (msirek) [inline db-engine/src/main/java/com/splicemachine/db/iapi/stats/ColumnStatisticsImpl.java:154]: Is this the final version of the fix?  It doesn't look like the serialization issue is fixed yet.

**Review by jyuanca — CHANGES_REQUESTED** [review]
This PR contains many commits that are not relevant to DB-11345. Can you clean it up so that it is easier to review?

**Pull request closed without merging**